Parser/asdl_c.py

```diff
@@ -947,8 +963,12 @@ def visitProduct(self, prod, name):
             fields = name+"_fields"
         else:
             fields = "NULL"
-        self.emit('state->%s_type = make_type("%s", state->AST_type, %s, %d);' %
-                        (name, name, fields, len(prod.fields)), 1)
+        self.emit(
+            'state->%s_type = make_type("%s", state->AST_type, %s, %d, "%s");'
+            % (name, name, fields, len(prod.fields), asdl_of(name, prod)),
```

Comment:
Ah, now I understand why you used `\\n` instead of `\n`. But I would use `\n` in asdl_py attributes and add a function which formats a string as a C literal (in particularly replaces `\n` -> `\\n"\n"` for readability and because not all compilers support long lines).

Reply:
Ah, thanks. I was looking for a way to split that strings into multiple lines but then forgot it. Will do.